public/app/plugins/datasource/prometheus/querybuilder/parsing.ts

```diff
@@ -95,6 +101,30 @@ interface Context {
   errors: ParsingError[];
 }
 
+function isValidPromQLMinusGrafanaGlobalVariables(expr: string) {
+  const context: Context = {
+    query: {
+      metric: '',
+      labels: [],
+      operations: [],
+    },
+    errors: [],
+  };
+
+  expr = expr.replace(/\$__interval/g, '1m');
+  expr = expr.replace(/\$__rate_interval/g, '1m');
```

Comment:
We also have the `$__range`. I do not know where exactly we use it but we have.
See: https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/components/monaco-query-field/monaco-completion-provider/completions.ts#L61

Reply:
It works but I do not feel safe with this. What if we have more variables in the future? What if we have variables that are not handled here? We must find a better approach I believe.

Reply:
Before sending the query to Prometheus, Grafana  is extrapolating the custom variables. There is probably the code somewhere in the backend. This backend probably hold the list of Global variables. We need  a way for the frontend to access this list of variables identifier, alongside a dummy value compatible with the type of that variable. So people adding global variables later will not have to update the frontend.

Do you think it is worth investigating?

Reply:
Added all variables I could find, with default values. 

Reading Loki and Prometheus plugins code, I can see that those variables are hard coded in multiples places. It does not seem to exist a single source of truth for what global variables Grafana provides and what default values should be used for syntax validation purpose.